src/android/org/apache/cordova/plugin/sentinel/CordovaPluginSentinelSDK.kt

```diff
@@ -0,0 +1,76 @@
+package org.apache.cordova.plugin
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.apache.cordova.CallbackContext
+import org.apache.cordova.CordovaPlugin
+import org.json.JSONArray
+import org.koin.core.context.GlobalContext.startKoin
+import org.koin.dsl.module
+
+import com.qxdev.sentinel_sdk.cloud.data.auth.User
+import com.qxdev.sentinel_sdk.cloud.data.auth.UserData
+import com.qxdev.sentinel_sdk.cloud.data.Response
+import com.qxdev.sentinel_sdk.di.Koin
+import com.qxdev.sentinel_sdk.SentinelProvider
+
+import org.apache.cordova.plugin.modules.AuthModule
+import org.apache.cordova.plugin.modules.LocationModule
+
+
+class CordovaPluginSentinelSDK : CordovaPlugin() {
+
+   private val authModule = AuthModule()
+   private val locationModule = LocationModule()
+
+   init {
+      startKoin { 
+         modules(
+            Koin.sentinelSDKModule("https://api-stage.sensys-iot.com")
+         ) 
+      }
+   }
+
+   override fun execute(
+         action: String,
+         args: JSONArray,
+         callbackContext: CallbackContext
```

Comment:
What about promises, can't we handle promises here?

Reply:
execute method just override callbacks

Done